EIPS/eip-5615.md

```diff
@@ -13,33 +13,36 @@ requires: 1155
+This EIP standardizes an existing mechanism to fetch token supply data from [EIP-1155](./eip-1155.md) tokens. It adds a `totalSupply` function, which fetches the number of tokens with a given `id`, and an `exists` function, which checks for the existence of a given `id`.
 
 ## Specification
 
 The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
 
 ```solidity
-interface ERC1155Supply /* is ERC1155 */ {
-  // @notice      This function MUST returns the total supply (number of tokens in circulation) with a given ID if it exists, otherwise it MUST revert
-  // @param   id  The ID of which fetch the total supply
-  // @return      The total supply of the given token ID
-  function totalSupply(uint256 id) external view returns (uint256);
-
-  // @notice      This function MUST return true if the given token ID exists, otherwise it MUST return false
-  // @param   id  The ID of which to check the existence
-  // @return      Whether the given token ID exists or not
+interface ERC1155Supply is ERC1155 {
+  // @notice      This function MUST return whether the given token id exists, previously existed, or may exist
+  // @param   id  The token id of which to check the existence
+  // @return      Whether the given token id exists, previously existed, or may exist
   function exists(uint256 id) external view returns (bool);
+
+  // @notice      If the given id exists, previously existed, or may exist, this function MUST return the number of tokens with a given id. Otherwise, this function MUST revert.
+  // @dev         This MUST revert if exists(id) returns false. This function therefore MUST call the exists function.
+  // @param   id  The token id of which fetch the total supply
+  // @return      The total supply of the given token id
+  function totalSupply(uint256 id) external view returns (uint256);
 }
 ```
 
 ## Rationale
 
-This EIP does not extend [EIP-165](./eip-165.md) as this interface was deemed simple enough that the extra complexity was unnecessary. It would also break backward compatibility with potential existing implementations.
+This EIP does not implement [EIP-165](./eip-165.md), as this interface is simple enough that the extra complexity is unnecessary and would cause incompatibilities with pre-existing implementations.
+
+For extensibility, `totalSupply` is required to call the `exists` function in case a contract extends the base implementation and overrides `exists`.
 
 ## Backwards Compatibility
 
-This EIP is backward compatible with the OpenZeppelin `ERC1155Supply`.
+No backward compatibility issues were found.
```

Comment:
[optional] When an EIP was designed specifically to compatible with some other EIP / protocol  in mind, i think it's great to call it out. I love the old sentence better
```suggestion
This EIP is designed to be backward compatible with the OpenZeppelin `ERC1155Supply`.
```